index.js

```diff
@@ -26,7 +26,7 @@ function _fillMemDuplex(memDuplexes, index, dataRetrievalFn, response, logger){
         (err, readable) => {
         if(err){
             logger.error('failed to get full object');
-            return response.end();
+            throw new Error('failed to get full object');
```

Comment:
Actually I like the error throwing because it will cause the process to crash (I hope) and fail with some connection-reset-by-peer error on the client. Otherwise I'm afraid the client would not really see the error and just get an invalid size situation or hang or ...